src/Altinn.Common/Altinn.EFormidlingClient/Altinn.EFormidlingClient/EFormidlingClient.cs

```diff
@@ -0,0 +1,381 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Altinn.Common.EFormidlingClient.Configuration;
+using Altinn.Common.EFormidlingClient.Models;
+using Altinn.Common.EFormidlingClient.Models.SBD;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Altinn.Common.EFormidlingClient
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IFormidlingClient"/> using a HttpClient.
+    /// </summary>
+    public class EFormidlingClient : IEFormidlingClient
+    {
+        private readonly HttpClient _client;
+        private readonly ILogger<EFormidlingClient> _logger;
+        private readonly IOptions<EFormidlingClientSettings> _appSettings;
+
+        /// <summary>
+        /// Initializes a new instance of the IFormidlingClient class with the given HttpClient, lSettings and Logger.
+        /// </summary>
+        /// <param name="client">A HttpClient provided by a HttpClientFactory.</param>
+        /// <param name="appSettings">The settings configured for eFormidling package</param>
+        /// <param name="logger">Logging</param>
+        public EFormidlingClient(HttpClient client, IOptions<EFormidlingClientSettings> appSettings, ILogger<EFormidlingClient> logger = null)
```

Comment:
maybe we could call them eformidlingSettings as not to confuse the app developers that are using the client?